aten/src/ATen/native/cuda/ForeachBinaryOpScalarList.cu

```diff
@@ -66,8 +67,49 @@ std::vector<Tensor> foreach_tensor_##NAME##_scalarlist_kernel_cuda(TensorList te
 }
 
 FOREACH_BINARY_OP_SCALARLIST(add, std::plus);
-FOREACH_BINARY_OP_SCALARLIST(sub, std::minus);
 FOREACH_BINARY_OP_SCALARLIST(mul, std::multiplies);
-FOREACH_BINARY_OP_SCALARLIST(div, std::divides);
+
+// In the case of division, integer inputs will result in float. 
+// Currently multi tensor apply can only return result of the same type as input.
+void foreach_tensor_div_scalarlist_kernel_cuda_(TensorList tensors, at::ArrayRef<Scalar> scalars) {
+    check_foreach_api_restrictions(tensors, scalars);
+    if (!can_use_fast_route(tensors, scalars, /*div_op*/ true)) {
+        return at::native::foreach_tensor_div_scalarlist_kernel_slow_(tensors, scalars);
+    }
+
+    foreach_binary_op_<std::divides>(tensors, scalars);
+}
+
+std::vector<Tensor> foreach_tensor_div_scalarlist_kernel_cuda(TensorList tensors, at::ArrayRef<Scalar> scalars) {
+    check_foreach_api_restrictions(tensors, scalars);
+    if (!can_use_fast_route(tensors, scalars, /*div_op*/ true)) {
+        return at::native::foreach_tensor_div_scalarlist_kernel_slow(tensors, scalars);
+    }
+
+    return foreach_binary_op<std::divides>(tensors, scalars);
+}
+
+// In the case of subtraction, we dont allow scalar to be boolean following the torch.sub logic
+void foreach_tensor_sub_scalarlist_kernel_cuda_(TensorList tensors, at::ArrayRef<Scalar> scalars) {
+    check_foreach_api_restrictions(tensors, scalars);
+    sub_check(tensors[0], scalars[0]);
+
+    if (!can_use_fast_route({tensors}, scalars)) {
+        return at::native::foreach_tensor_sub_scalarlist_kernel_slow_(tensors, scalars);
+    }
+
+    foreach_binary_op_<std::minus>(tensors, scalars);
+}
+
+std::vector<Tensor> foreach_tensor_sub_scalarlist_kernel_cuda(TensorList tensors, at::ArrayRef<Scalar> scalars) {
+    check_foreach_api_restrictions(tensors, scalars);
+    sub_check(tensors[0], scalars[0]);
```

Comment:
It's not guaranteed that all of the scalars have the same dtype, right? Can a user do the following?

```
tensors = [torch.ones([], dtype=torch.bool) for _ in range(3)]
scalars = [1, 1, True]
_foreach_sub(tensors, scalars)
```

Reply:
Scalars of different dtype are allowed but sub will throw an error in case of boolean scalar